### PR TITLE
Kubernetes/plugin site

### PR DESCRIPTION
**Pull request by olblak**
This PR deploy plugins.jenkins.io and its requirements (lego/nginx-ingress)

1) [Ingress|https://kubernetes.io/docs/user-guide/ingress/]
2) [Kube-lego| https://github.com/jetstack/kube-lego]
3) Plugin-site

Once everything is deployed, we have to setup a dns record to the ingress public ip.
Use `kubectl get service nginx --namespace nginx-ingress` to retrieve public ip used by ingress

By default plugin-site use plugins.jenkins.io
We can override it with  profile::kubernetes::resources::pluginsjenkins::url=<your_url>

### CODE REVIEW

**Comment by rtyler** [comment]
@olblak Can you rebase this branch?

File: dist/profile/templates/kubernetes/resources/pluginsjenkins/deployment.yaml.erb [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        metadata:
+            labels:
+                app: plugins-jenkins
+                type: plugins-jenkins
+                logtype: archive
+                configmap-plugins-jenking: linked
+        spec:
+            containers:
+                - name: plugins-jenkins
+                  image: jenkinsciinfra/plugin-site:50-8a166d
```
Comment (rtyler) [inline dist/profile/templates/kubernetes/resources/pluginsjenkins/deployment.yaml.erb]: This should be pulled from hiera (it's already in there)

**Review by rtyler — APPROVED** [review]
This looks good to me, are we ready to deploy to production @olblak ?

**Comment by olblak** [comment]
May the god of Microservices be with us

**Pull request merged**